pkg/provider/gitlab/gitlab.go

```diff
@@ -155,10 +156,28 @@ func (g *Gitlab) NewClient(ctx context.Context, store esv1beta1.GenericStore, ku
 	return g, nil
 }
 
-// Empty GetAllSecrets.
+// GetAllSecrets syncs all gitlab project variables into a single Kubernetes Secret.
 func (g *Gitlab) GetAllSecrets(ctx context.Context, ref esv1beta1.ExternalSecretFind) (map[string][]byte, error) {
-	// TO be implemented
-	return nil, fmt.Errorf("GetAllSecrets not implemented")
+	if utils.IsNil(g.client) {
+		return nil, fmt.Errorf(errUninitializedGitlabProvider)
+	}
```

Comment:
Also, if finding by tags is not supported, we should check if the `reg.Tags` is set, and properly error out with a message similar to  `dataFrom.find.tags is not supported for gitlab provider`.

Reply:
done